ui-components/menuitems.md

```diff
@@ -0,0 +1,124 @@
+* Create a button for custom behavior.
+
+## How to create custom menu items
+
+The methods for adding custom menu items are in the UI Registry part of the editor API editor.ui.registry. The API has three methods for adding menu items:
+
+* `editor.ui.registry.addMenuItem`(identifier, configuration)
+* `editor.ui.registry.addToggleMenuItem`(identifier, configuration)
+* `editor.ui.registry.addNestedMenuItem`(identifier, configuration)
+
+The two arguments these methods take are:
+
+* `identifier` - a unique name for the button
+* `configuration` - an object containing your configuration for that button.
+
+Define the custom toolbar button with the `setup` callback of the TinyMCE configuration to add it to the editor. This callback is invoked automatically for every initialized editor instance. Access to the UI registry API occurs when the callback receives a reference to the editor instance as its argument.
+
+### Example of adding a custom menu item
+
+```js
+tinymce.init({
+  selector: '#editor',
+  menu: {
+    custom: { title: 'Custom Menu', items: 'undo redo myCustomMenuItem' }
+  },
+  menubar: 'file edit custom',
+  setup: (editor) => {
+    editor.ui.registry.addMenuItem('myCustomMenuItem', {
+      text: 'My Custom Menu Item', 
+      onAction: () => alert('Menu item clicked')
+    });
+  }
+});
+
+```
+
+> Note: The identifier used to create the menu item must be included in the [`menu`](https://www.tiny.cloud/docs-beta/configure/editor-appearance/#menu) option in the TinyMCE configuration for it to be added to the menubar's menus. It will not be added to the menubar's menus if `menu` is not configured correctly.
+
+
+## Types of menu items
+
+There are three types of menu items in TinyMCE 5.0:
+
+* Basic menu items
+* Nested menu items
+* Toggle menu items
+
+### Basic menu items
+
+A basic menu item triggers its `onAction` function when clicked.
+
+#### Config options
+
+| Name | Value | Requirement | Description |
+| ---- | ----- | ----------- | ----------- |
+| text | string | optional | Text to display if no icon is found. |
+| icon | string | optional | Name of the icon to be displayed. Must correspond to an icon in the icon pack. |
+| value | string | optional | A value to associate with the menu item. |
+| disabled | boolean | optional | default: false - Represents the menu item's state. When true, the menu item is unclickable. Toggled by the menu item's API. |
+| onSetup | (api) => (api) => void | optional | default: () => () => {} - Function invoked when the menu item is rendered, each time its menu is opened. |
+| onAction | (api) => void | required | Function invoked when the menu item is clicked. |
+
+#### API
+
+| Name | Value | Description |
+| ---- | ----- | ----------- |
+| isDisabled | () => boolean | Checks if the menu item is disabled. |
+| setDisabled | (state: boolean) => void | Sets the menu item's disabled state. |
+
+### Nested menu items
```

Comment:
I wrote this very quickly, hence there aren't any examples for the nested or toggle menu item sections. We should really add some, even if they're very simple. Could even take the `editor.ui.registry.add...` examples from the demo page and throw them in here?

Reply:
:+1: I largely reused them, though made them use alerts instead of inserting content.